packages/nexus/src/events.ts

```diff
@@ -0,0 +1,124 @@
+import { z } from 'zod'
+import { AtUri } from '@atproto/syntax'
+
+export const recordEventDataSchema = z.object({
+  did: z.string(),
+  rev: z.string(),
+  collection: z.string(),
+  rkey: z.string(),
+  action: z.enum(['create', 'update', 'delete']),
+  record: z.record(z.string(), z.unknown()).optional(),
+  cid: z.string().optional(),
+})
+export type RecordEventData = z.infer<typeof recordEventDataSchema>
+
+export const userEventDataSchema = z.object({
+  did: z.string(),
+  handle: z.string(),
+  is_active: z.boolean(),
+  status: z.enum([
+    'active',
+    'takendown',
+    'suspended',
+    'deactivated',
+    'deleted',
+  ]),
+})
+export type UserEventData = z.infer<typeof userEventDataSchema>
+
+export const recordEvt = z.object({
+  id: z.number(),
+  type: z.literal('record'),
+  record: recordEventDataSchema,
+})
+
+export const userEvt = z.object({
+  id: z.number(),
+  type: z.literal('user'),
+  user: userEventDataSchema,
+})
+
+export const nexusEvt = z.union([recordEvt, userEvt])
+
+export interface NexusEvent {
+  id: number
+  did: string
+  isRecordEvt(): this is RecordEvent
+  isUserEvt(): this is UserEvent
```

Comment:
Just a style thing, but I'd advocate lightly for `isRecordEvent` and `isUserEvent`.